cmd/influxd/run.go

```diff
@@ -306,6 +306,14 @@ func (cmd *RunCommand) CheckConfig() {
 func (cmd *RunCommand) Open(config *Config, join string) *Node {
 	if config != nil {
 		cmd.config = config
+
+		// Override config properties.
+		if cmd.hostname != "" {
```

Comment:
This check is already done here: https://github.com/runner-mei/influxdb/blob/hostname_is_empty/cmd/influxd/run.go#L213

The reason it is doe there is to allow testing to specify whatever it needs, this would break our ability to test properly.